### PR TITLE
Rbd disk type support.

### PR DESCRIPTION
**Pull request by rbian**
libvirt_storage, virsh: add rbd pool support; modify pool_define_as function to make target as the optional
disk, pool_xml, secret_xml: add some new rbd pool/volume attributes to xml.

### CODE REVIEW

File: virttest/virsh.py [hunk 1]
```diff
@@ -2009,7 +2009,7 @@ def pool_state_dict(only_names=False, **dargs):
     return result
 
 
-def pool_define_as(name, pool_type, target, extra="", **dargs):
+def pool_define_as(name, pool_type, target="", extra="", **dargs):
     """
```
Comment (will-Do) [inline virttest/virsh.py]: I saw the func doc mentioned pool type, so you can also update it:
1). typo: s/typ/type
2). Add 'rbd' type desc

Comment (rbian) [inline virttest/virsh.py]: Thanks @will-Do , I have updated the patch.

**Comment by will-Do** [comment]
ACK

**Comment by waynesun09** [comment]
ACK

**Pull request merged**